### PR TITLE
Move make offset map to load stage

### PR DESCRIPTION
**Pull request by foxspy**
issue: #132 
Delete Making map at `getVectors` will cause unexpected error for existing ivf index, because these indexes have no `direct_map` in the binary set. So just move the making operation to load stage to support `getVectors` from existing indexes.

### CODE REVIEW

File: src/index/ivf/ivf.cc [hunk 1]
```diff
@@ -320,15 +319,13 @@ IvfIndexNode<T>::Train(const DataSet& dataset, const Config& cfg) {
             index = std::make_unique<faiss::IndexIVFScalarQuantizer>(qzr, dim, nlist, faiss::QuantizerType::QT_8bit,
                                                                      metric.value());
             index->train(rows, (const float*)data);
-            index->make_direct_map(true);
```
Comment (liliu-z) [inline src/index/ivf/ivf.cc:323]: We don't need this for SQ and PQ

Comment (foxspy) [reply src/index/ivf/ivf.cc:323]: limited the make operation by having raw data or not

**Pull request merged**